code/modules/surgery/bodyparts/parts.dm

```diff
@@ -87,6 +87,48 @@
 	unarmed_damage_high = 10
 	unarmed_stun_threshold = 10
 	body_zone = BODY_ZONE_L_ARM
+	/// Offsets for held items. If your hand is shorter and you wanna move the items up.
+	var/held_offset = 0
+	/// Are we covered with blood?
+	var/bloody = FALSE
+	/// Icon_state for the bloody hand.
+	var/bloody_icon
+
+/obj/item/bodypart/arm/update_limb(dropping_limb, is_creating)
+	. = ..()
+	var/mob/living/carbon/human/human_owner = owner
+	bloody = istype(human_owner) && human_owner.blood_in_hands
+
+/obj/item/bodypart/arm/generate_icon_key()
+	. = ..()
+	if(bloody)
+		. += "-bloody"
+	return .
+
+/obj/item/bodypart/arm/get_limb_icon(dropped)
+	. = ..()
+	if(bloody)
+		. += image('icons/effects/blood.dmi', bloody_icon, -GLOVES_LAYER, pixel_y = y_offset)
+	return .
+
+/obj/item/bodypart/arm/set_disabled(new_disabled)
+	. = ..()
+	if(isnull(.) || !owner)
+		return
```

Comment:
can't remember if this needs to be return . or not

Reply:
@Mothblocks

Reply:
Dunno about "needs", the rules are intentionally ambiguous, but if the value of `.` is actually important enough to store here then I would personally do `return .`